src/Components/test/testassets/Components.TestServer/BlazorWebServerStartup.cs

```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using Components.TestServer.RazorComponents;
+using Microsoft.AspNetCore.Components.Server.Circuits;
+using Microsoft.AspNetCore.Components.Web;
+using Microsoft.AspNetCore.DataProtection;
+
+namespace TestServer;
+
+public class BlazorWebServerStartup
+{
+    public BlazorWebServerStartup(IConfiguration configuration)
+    {
+        Configuration = configuration;
+    }
+
+    public IConfiguration Configuration { get; }
+
+    // This method gets called by the runtime. Use this method to add services to the container.
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddRazorComponents()
+            .AddServerComponents(options =>
+            {
+                options.RootComponents.MaxJSRootComponents = 5; // To make it easier to test
+                options.RootComponents.RegisterForJavaScript<BasicTestApp.DynamicallyAddedRootComponent>("my-dynamic-root-component");
+                options.RootComponents.RegisterForJavaScript<BasicTestApp.JavaScriptRootComponentParameterTypes>(
+                    "component-with-many-parameters",
+                    javaScriptInitializer: "myJsRootComponentInitializers.testInitializer");
+            });
+        services.AddSingleton<ResourceRequestLog>();
+        services.AddTransient<BasicTestApp.FormsTest.ValidationComponentDI.SaladChef>();
```

Comment:
Is all this stuff really used? If any of this is not actually required for the tests we're doing against this app, let's be sure to remove it otherwise it's super confusing.

If there's a reasonable way to avoid adding a new startup class that would be better still, because we don't have any system for knowing which test corresponds to which startup class, so when you're trying to debug them it's quite difficult to know what entry in `Components.TestServer/Program.cs` to click on.